### PR TITLE
[RHOAIENG-14687] Extend tests for certificate to assure that they are readable

### PR DESCRIPTION
**Pull request by jstourac**
https://issues.redhat.com/browse/RHOAIENG-14687

## How Has This Been Tested?

In `components/odh-notebook-controller` run `make test`.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: components/odh-notebook-controller/controllers/notebook_controller_test.go [hunk 1]
```diff
@@ -1039,3 +1053,32 @@ func createOAuthConfigmap(name, namespace string, label map[string]string, confi
...
+
+// checkCertConfigMap checks the content of a config map defined by the name and namespace
+// It triest to parse the given certFileName and checks that all certificates can be parsed there and that the number of the certificates matches what we expect.
+func checkCertConfigMap(ctx context.Context, namespace string, configMapName string, certFileName string, expNumberCerts int) {
+	configMap := &corev1.ConfigMap{}
+	key := types.NamespacedName{Namespace: namespace, Name: configMapName}
+	Expect(cli.Get(ctx, key, configMap)).Should(Succeed())
+
+	// Attempt to decode PEM encoded certificates so we are sure all are readable as expected
+	certData, _ := configMap.Data[certFileName]
```
Comment (jiridanek) [inline components/odh-notebook-controller/controllers/notebook_controller_test.go:1065]:
```suggestion
	certData := configMap.Data[certFileName]
```

that is what the linter wants you to do; fetching from a map has a single valued result and two valued result, and go will pick depending on how many params you are taking out

https://go.dev/tour/moretypes/22

Comment (jstourac) [reply components/odh-notebook-controller/controllers/notebook_controller_test.go]: You're right, thanks, updated :slightly_smiling_face:

**Review by jiridanek — COMMENTED** [review]
/lgtm

was around while this was being made, so no complaints from me

**Comment by jstourac** [comment]
The e2e tests are failing due to an infra issue and are irrelevant for changes in this PR anyway, so overriding:

/override ci/prow/odh-notebook-controller-e2e

**Comment by atheo89** [comment]
/lgtm

Also today the e2e is working!

**Comment by jstourac** [comment]
Thank you for the review, @atheo89 

> Also today the e2e is working!

well, if you're judging only based on the result here - actually I overrode them as they are irrelevant for this change anyway :angel:

**Comment by jstourac** [comment]
Thank you all. I'm gonna merge this to move on.

/approve

**Pull request merged**